lib/posticle/src/grammar.pest

```diff
@@ -0,0 +1,121 @@
+    | ^"xmlrpc.beeps" | ^"simpleledger" | ^"gizmoproject" | ^"ms-officeapp" | ^"ms-getoffice"
+    | ^"ms-publisher" | ^"onenote-cmd" | ^"ms-drive-to" | ^"ms-settings" | ^"thismessage"
+    | ^"xmlrpc.beep" | ^"bitcoincash" | ^"xcon-userid" | ^"view-source" | ^"ms-infopath"
+    | ^"ms-inputapp" | ^"openpgp4fpr" | ^"soap.beeps" | ^"ms-walk-to" | ^"ms-project" | ^"secondlife"
+    | ^"mailserver" | ^"browserext" | ^"attachment" | ^"filesystem" | ^"coaps+tcp" | ^"soap.beep"
+    | ^"feedready" | ^"ms-people" | ^"iris.beep" | ^"adiumxtra" | ^"teamspeak" | ^"iris.xpcs"
+    | ^"hydrazone" | ^"paparazzi" | ^"ms-access" | ^"ms-excel" | ^"ventrilo" | ^"isostore"
+    | ^"iris.xpc" | ^"iris.lwz" | ^"iotdisco" | ^"teliaeid" | ^"coap+tcp" | ^"videotex" | ^"coaps+ws"
+    | ^"resource" | ^"prospero" | ^"ms-visio" | ^"facetime" | ^"diaspora" | ^"platform" | ^"example"
+    | ^"bitcoin" | ^"keyparc" | ^"appdata" | ^"session" | ^"ms-help" | ^"wyciwyg" | ^"coap+ws"
+    | ^"service" | ^"z39.50s" | ^"v-event" | ^"spotify" | ^"message" | ^"onenote" | ^"mupdate"
+    | ^"ms-word" | ^"content" | ^"z39.50r" | ^"beshare" | ^"mongodb" | ^"mailto" | ^"webcal"
+    | ^"lastfm" | ^"jabber" | ^"gopher" | ^"z39.50" | ^"ms-spd" | ^"finger" | ^"mumble" | ^"pkcs11"
+    | ^"market" | ^"rediss" | ^"reload" | ^"magnet" | ^"chrome" | ^"soldat" | ^"spiffe" | ^"callto"
+    | ^"submit" | ^"telnet" | ^"things" | ^"barion" | ^"tn3270" | ^"unreal" | ^"ut2004" | ^"rsync"
+    | ^"ldaps" | ^"ymsgr" | ^"rtmfp" | ^"rtsps" | ^"rtspu" | ^"conti" | ^"coaps" | ^"shttp" | ^"sieve"
+    | ^"skype" | ^"snews" | ^"about" | ^"hxxps" | ^"msnim" | ^"msrps" | ^"steam" | ^"stuns" | ^"https"
+    | ^"notes" | ^"gtalk" | ^"graph" | ^"modem" | ^"turns" | ^"query" | ^"redis" | ^"vemmi" | ^"proxy"
+    | ^"xfire" | ^"ldap" | ^"hxxp" | ^"psyc" | ^"xmpp" | ^"maps" | ^"http" | ^"hcap" | ^"h323"
+    | ^"crid" | ^"lvlt" | ^"itms" | ^"rtmp" | ^"rtsp" | ^"iris" | ^"ircs" | ^"irc6" | ^"ipps"
+    | ^"coap" | ^"sftp" | ^"fish" | ^"info" | ^"sips" | ^"file" | ^"smtp" | ^"feed" | ^"snmp"
+    | ^"elsi" | ^"ed2k" | ^"imap" | ^"msrp" | ^"aaas" | ^"stun" | ^"mtqp" | ^"pwid" | ^"bolo"
+    | ^"blob" | ^"dntp" | ^"tftp" | ^"news" | ^"nntp" | ^"icon" | ^"tool" | ^"turn" | ^"dict"
+    | ^"acap" | ^"pack" | ^"palm" | ^"acct" | ^"icap" | ^"wais" | ^"data" | ^"wpid" | ^"pres"
+    | ^"wtai" | ^"xcon" | ^"jms" | ^"ham" | ^"smb" | ^"sms" | ^"dav" | ^"fax" | ^"jar" | ^"pop"
+    | ^"cap" | ^"irc" | ^"cvs" | ^"dvb" | ^"ssh" | ^"dtn" | ^"mid" | ^"iax" | ^"ipp" | ^"svn"
+    | ^"tag" | ^"git" | ^"tel" | ^"ipn" | ^"geo" | ^"res" | ^"mvn" | ^"rmi" | ^"tip" | ^"ftp"
+    | ^"nfs" | ^"xri" | ^"nih" | ^"moz" | ^"udp" | ^"mms" | ^"urn" | ^"ocf" | ^"oid" | ^"dns"
+    | ^"apt" | ^"aim" | ^"vnc" | ^"afs" | ^"dis" | ^"sgn" | ^"did" | ^"hcp" | ^"wss" | ^"cid"
+    | ^"afp" | ^"sip" | ^"acr" | ^"aaa" | ^"ws" | ^"aw" | ^"tv" | ^"qb" | ^"ni" | ^"gg" | ^"go"
+    | ^"im") ~ "://"
```

Comment:
a uri does not need the `//` if it does not contain an authority component, like with `mailto:` for instance

Reply:
I think we should go for matching URLs only for the time being as that is what the existing code does and changing that means work needs to be done on rustodon itself. We'll get there, but if it's ok with everyone I'll do it in a separate PR.

For the time being I'll try and find or make a definitive list of URL schemas, so we won't match `mailto://`.

Reply:
i'm perfectly fine with a naive matcher, just with the schema, a colon, and any valid characters after it